app/src/main/java/org/p2p/wallet/rpc/interactor/CloseInteractor.kt

```diff
@@ -0,0 +1,37 @@
+package org.p2p.wallet.rpc.interactor
+
+import org.p2p.solanaj.core.Account
+import org.p2p.solanaj.core.Transaction
+import org.p2p.solanaj.programs.TokenProgram
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.rpc.repository.blockhash.RpcBlockHashRepository
+import org.p2p.wallet.rpc.repository.history.RpcHistoryRepository
+import org.p2p.wallet.utils.toPublicKey
+
+class CloseInteractor(
+    private val rpcBlockHashRepository: RpcBlockHashRepository,
+    private val rpcTransactionRepository: RpcHistoryRepository,
+    private val tokenKeyProvider: TokenKeyProvider
+) {
+
+    suspend fun close(addressToClose: String): String {
+        val owner = tokenKeyProvider.publicKey.toPublicKey()
+        val instruction = TokenProgram.closeAccountInstruction(
+            TokenProgram.PROGRAM_ID,
+            addressToClose.toPublicKey(),
+            owner,
```

Comment:
named args please, two identical args

Reply:
For now we have limitation because of Java, to add support for naming args we need to refactor TokenProgram to kotlin class